app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -375,6 +378,74 @@ public void onClick(View v) {
         streamActv.setAdapter(streamActvAdapter);
         topicActv.setAdapter(subjectActvAdapter);
         checkAndSetupStreamsDrawer();
+        SimpleCursorAdapter combinedAdapter = new SimpleCursorAdapter(
+                that, R.layout.emoji_tile, null,
+                new String[]{Emoji.NAME_FIELD, Emoji.NAME_FIELD},
+                new int[]{R.id.emojiImageView, R.id.nameTV}, 0);
+
+        combinedAdapter.setCursorToStringConverter(new SimpleCursorAdapter.CursorToStringConverter() {
+            @Override
+            public CharSequence convertToString(Cursor cursor) {
+                if (cursor == null) return messageEt.getText();
+                int index = cursor.getColumnIndex(Emoji.NAME_FIELD);
+                String name = cursor.getString(index);
+                String currText = messageEt.getText().toString();
+                int last = (cursor.getColumnIndex(Emoji.NAME_FIELD) == 6) ? currText.lastIndexOf("@") : currText.lastIndexOf(":");
+                return TextUtils.substring(currText, 0, last) + ((cursor.getColumnIndex(Emoji.NAME_FIELD) == 6) ? "@**" + name + "**" : ":" + name.replace(".png", "") + ":");
+            }
+        });
+        combinedAdapter.setFilterQueryProvider(new FilterQueryProvider() {
+            @Override
+            public Cursor runQuery(CharSequence charSequence) {
+                if (charSequence == null) return null;
+                int length = charSequence.length();
+                int personLength = charSequence.toString().lastIndexOf("@");
+                int smileyLength = charSequence.toString().lastIndexOf(":");
+                if (length - 1 > Math.max(personLength, smileyLength) + MAX_THRESOLD_EMOJI_HINT
+                        || length - Math.max(personLength, smileyLength) - 1 < MIN_THRESOLD_EMOJI_HINT
+                        || (personLength + smileyLength == -2))
+                    return null;
+                try {
+                    if (personLength > smileyLength) {
+                        return makePeopleNameCursor(charSequence.subSequence(personLength + 1, length));
+                    } else {
+                        return makeEmojiCursor(charSequence.subSequence(smileyLength + 1, length));
+                    }
+                } catch (SQLException e) {
+                    Log.e("SQLException", "SQL not correct", e);
+                    return null;
+                }
+            }
+        });
+        messageEt.setAdapter(combinedAdapter);
+    }
+
+    private Cursor makeEmojiCursor(CharSequence emoji)
+            throws SQLException {
+        if (emoji == null) {
+            emoji = "";
+        }
+        return ((AndroidDatabaseResults) app
+                .getDao(Emoji.class)
+                .queryRaw("SELECT  rowid _id,* FROM emoji WHERE name LIKE '%" + emoji + "%'")
```

Comment:
It's best practice to name the fields you're expecting to return, instead of using the `*` glob pattern.